sklearn/metrics/classification.py

```diff
@@ -1912,8 +1912,21 @@ def brier_score_loss(y_true, y_prob, sample_weight=None, pos_label=None):
     assert_all_finite(y_true)
     assert_all_finite(y_prob)
 
+    # currently, we only support binary classification
+    _check_binary_probabilistic_predictions(y_true, y_prob)
+
+    # ensure valid y_true if pos_label is not specified
+    classes = np.unique(y_true)
```

Comment:
I meant that this logic could move to the helper, so that other binary probabilistic metrics would similarly identify -1 alone as the negative class, and to avoid rubbing unique multiple times.

Reply:
@jnothman Sorry but I don't quite understand. This part is about giving pos_label the right value and raise an error when needed. Currently,  `_check_binary_probabilistic_predictions` don't accept pos_label as a parameter. So it might be hard to move the logic into `_check_binary_probabilistic_predictions` .

Reply:
After determining `pos_label`'s appropriate value, it is used once: to binarise `y_true` into {0,1}. This is also what `_check_binary_probabilistic_predictions` does, only without the ability to configure `pos_label`. If `_check_binary_probabilistic_predictions` took an optional `pos_label` you could handle all cases.

Reply:
@jnothman Sorry to disturb again but I have some questions:
(1)It seems that `_check_binary_probabilistic_predictions` is only used in brier_score_loss(metrics.classification) and calibration_curve(calibration).
The latter do not support pos_label.
(2)I am wondering how we should took the optional pos_label since even if we support pos_label, there are cases when pos_label is not provided by users(default to None).
It seems that I cannot use `def(..., pos_label=None)` and it seems ugly to use something like `def(..., **dict)` .
(3)From my perspective, since `_check_binary_probabilistic_predictions` is only used in two functions and in each of these two functions, some part of it is meaningless(e.g., label_binarize in brier_score_loss and the check of y_prob in calibration_curve), how about remove the function(no test directly use the function)?

Reply:
label_binarize isn't meaningless in brier: it's basically identical to what's happening in https://github.com/scikit-learn/scikit-learn/pull/9562/files/adfbbec50f326fe0a3678ea29c630269ef258fe1#diff-b04acd877dd793f28ae7be13a999ed88R1931

Reply:
And why can't you use `def _check_binary_probabilistic_predictions(y_true, y_prob, pos_label=None)`?

Reply:
@jnothman For example, in the function, if pos_label=None, there are two situations
(1)I don't pass pos_label(used in calibration)
(2)I pass pos_label=None(used in brier_score_loss when pos_label is not provided)
How can I distinguish them? Thanks. :)

Reply:
Why do you need to distinguish them? Should they not be treated the same?